lib/plug/ssl.ex

```diff
@@ -58,6 +87,40 @@ defmodule Plug.SSL do
   require Logger
   import Plug.Conn
 
+  @strong_tls_ciphers [
+    'DHE-RSA-AES256-GCM-SHA384',
+    'DHE-RSA-AES128-GCM-SHA256',
+    'ECDHE-RSA-AES256-GCM-SHA384',
+    'ECDHE-ECDSA-AES256-GCM-SHA384',
+    'ECDHE-RSA-AES128-GCM-SHA256',
+    'ECDHE-ECDSA-AES128-GCM-SHA256'
+  ]
```

Comment:
Hmm, odd that OWASP puts DHE above ECDHE, even though they note in a comment that "...the TLS handshake with DHE hinders the CPU about 2.4 times more than ECDHE". Every other source will tell you to put ECDHE before DHE for performance reasons. Let's take that liberty with the OWASP list, it really makes a difference in the connection rate a server can support.

Again, sorry for missing that last night...

Reply:
I thought it was odd too. I kept it though, because it's much simpler to say we follow OWASP.

In my config I use the order you're suggesting so I agree with the change but I think we better explain this in the readme then. To clarify why we're deviating from it.

Reply:
I am moving to this ordering then:

```elixir

  @strong_tls_ciphers [
    'ECDHE-RSA-AES256-GCM-SHA384',
    'ECDHE-ECDSA-AES256-GCM-SHA384',
    'ECDHE-RSA-AES128-GCM-SHA256',
    'ECDHE-ECDSA-AES128-GCM-SHA256',
    'DHE-RSA-AES256-GCM-SHA384',
    'DHE-RSA-AES128-GCM-SHA256'
  ]

  @compatible_tls_ciphers [
    'ECDHE-RSA-AES256-GCM-SHA384',
    'ECDHE-ECDSA-AES256-GCM-SHA384',
    'ECDHE-RSA-AES128-GCM-SHA256',
    'ECDHE-ECDSA-AES128-GCM-SHA256',
    'DHE-RSA-AES256-GCM-SHA384',
    'DHE-RSA-AES128-GCM-SHA256',
    'ECDHE-RSA-AES256-SHA384',
    'ECDHE-ECDSA-AES256-SHA384',
    'ECDHE-RSA-AES128-SHA256',
    'ECDHE-ECDSA-AES128-SHA256',
    'ECDHE-RSA-AES256-SHA',
    'ECDHE-ECDSA-AES256-SHA',
    'DHE-RSA-AES256-SHA256',
    'DHE-RSA-AES128-SHA256',
    'ECDHE-RSA-AES128-SHA',
    'ECDHE-ECDSA-AES128-SHA'
  ]

```